install_dependencies.sh

```diff
@@ -52,7 +55,8 @@ if echo $REPLY | grep -E '^[Yy]$' > /dev/null; then
                 sudo apt-get install --no-install-recommends -y software-properties-common lsb-release
 
                 # Packages added here should also be added to the Dockerfile
-                sudo apt-get install --no-install-recommends -y autoconf bash-completion bc clang-16 clang-17 clang-format-17 clang-tidy-17 cmake curl dos2unix g++-14 gcc-14 git graphviz libboost-all-dev libhwloc-dev libncurses5-dev libnuma-dev libnuma1 libpq-dev libreadline-dev libsqlite3-dev libtbb-dev lld-17 man parallel postgresql-server-dev-all python3 python3-pip valgrind &
+                sudo apt-get install --no-install-recommends -y git python3 python3-pip
```

Comment:
Please explain in a comment why we add certain tools before others. Also merge with line 55?